src/emeus-constraint-layout.c

```diff
@@ -2400,7 +2407,7 @@ emeus_constraint_layout_child_set_intrinsic_width (EmeusConstraintLayoutChild *c
 
   g_return_if_fail (EMEUS_IS_CONSTRAINT_LAYOUT_CHILD (child));
 
-  if (child->intrinsic_width == width)
+  if (child->solver == NULL || child->intrinsic_width == width)
```

Comment:
I usually like to separate different, unrelated conditions, instead of lumping them all together.

In this case, the `solver` being `NULL` is a separate failure condition from the child's intrinsic width being the same value as the passed width; thus, the conditions should be different, even if the result is the same.